server/channels/api4/role.go

```diff
@@ -47,7 +47,10 @@ func getAllRoles(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	w.Write(js)
+	if _, err := w.Write(js); err != nil {
+		c.Err = model.NewAppError("getAllRoles", "api.roles.get_roles.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		return
+	}
```

Comment:
Sending another error to the client if a write fails will most likely also fail. Let's just log an error, similar to https://github.com/pradeepmurugesan/mattermost/blob/8f53dfcb588b8c3e2cbb4b7e833544e5d674c784/server/channels/api4/channel.go#L131-L133